src/modules/auth/token.service.spec.ts

```diff
@@ -0,0 +1,85 @@
+import { Controller, Get, INestApplication, Injectable, InternalServerErrorException, MiddlewareConsumer, Module, RequestMethod, UnauthorizedException } from '@nestjs/common';
+import { Test, TestingModule } from '@nestjs/testing';
+import { Request } from 'express';
+import { NextFunction } from 'express-serve-static-core';
+import request from 'supertest';
+import * as jwt from 'jsonwebtoken';
+import { TokenPayload } from './token.service';
+
+// Mock token service class
+@Injectable()
+class TokenService {
+
+    handleOriginCheck = async (req: Request, res: Response, next: NextFunction) => {
+        let token = null;
+        if (req.headers['authorization']) {
+            token = req.headers['authorization'].replace('Bearer ', '');
+        } else {
+            token = req?.cookies?.token
+        }
+
+        if (token) {
+            const decodedToken = jwt.decode(token) as TokenPayload;
+            const isBrowserRequestFromAuthenticatedOrigin = decodedToken.origin === req.headers['origin']
+            const isServerRequestOrGETFromSameDomain = req.headers['origin'] === undefined
+            if (isBrowserRequestFromAuthenticatedOrigin || isServerRequestOrGETFromSameDomain) {
+                next()
+            } else {
+                throw new InternalServerErrorException('Something went wrong');
+            }
+        } else {
+            throw new UnauthorizedException('Unauthorized');
+        }
+    }
+}
```

Comment:
@hrachocode you wrote:
> the reason behind this is when we are using an imported function from TokenService we should also import all other dependencies and their configs including AuthModule for example. so this way we are creating a mock of that function rather then working with dependencies.

Sorry, I'm not understanding this at all 😅 . In this case, the `TokenService` is the class under test, so it definitely shouldn't be mocked. If the test is using a mock `TokenService` then how does the test tell us anything about `TokenService` that is actually used during runtime by the app?

Reply:
Ok in other words, when removing the Mock function and using `TokenService` class as a provider to our test module
it throws errors like 

`Nest can't resolve dependencies of the TokenService (?, JwtService, RefreshTokenRepository). Please make sure that the argument ConfigService at index [0] is available in the TestModule context.`

and when I am trying to import `ConfigService` in `TestModule` providers, then it requires `RefreshTOkenRepository`, after it requires `JwtService`

and after it triggers an error that `JWT_MODULE_OPTIONS` are not provided.

and this list is deeper as when I am trying to use real `TokenService class`  :/ 

that's why I decided to mock that function rather than importing all project configs in one test file.

Reply:
As I understand, mocking the dependencies of the class under test is the whole point of mock. We test the real class but mock its dependencies. So instead of importing the real `ConfigService` (which then requires other imports as you say), we instead mock `ConfigService` so that it doesn't need any imports

Reply:
yeah correct, but we are mocking `TokenService` not `ConfigService`